lib/miq_expression/tag.rb

```diff
@@ -1,17 +1,27 @@
-class MiqExpression::Tag
-  def self.parse(tag)
-    klass, ns = tag.split(".")
-    klass, ns = "nil", klass if ns.nil? # support managed-label
-    ns = "/" + ns.split("-").join("/")
-    ns = ns.sub(/(\/user_tag\/)/, "/user/") # replace with correct namespace for user tags
-    new(klass.safe_constantize, ns)
-  end
+class MiqExpression::Tag < MiqExpression::Field
+  TAG_REGEX = /
+(?<model_name>([[:alnum:]]*(::)?)+)
+\.(?<associations>([a-z_]+\.)*)
+(?<namespace>\bmanaged|user_tag\b)
```

Comment:
it looks like  we don't need it but we decided that we will keep it:
https://github.com/ManageIQ/manageiq/pull/13769#issuecomment-279079989